tutorials/2d/2d_lights_and_shadows.rst

```diff
@@ -0,0 +1,226 @@
+  * :ref:`Sprite <class_Sprite>`
+  * :ref:`Light2D <class_Light2D>`
+  * :ref:`LightOccluder2D <class_LightOccluder2D>`
+
+:ref:`CanvasModulate<class_CanvasModulate>` is used to add ambient light to the scene. Ambient 
+light is not actually necessary, it just darkens the scene. 
+
+:ref:`Sprites <class_Sprite>` are used to display the textures for the light blobs, the 
+background, and for the shadow casters.
+
+:ref:`Light2Ds <class_Light2D>` are used to light the scene. The way a light typically works 
+is by selecting a light texture and then adding that texture over the rest of the scene to 
+simulate lighting. 
+
+:ref:`LightOccluder2Ds <class_LightOccluder2D>` are used to tell the shader which parts of 
+the scene cast shadows. The shadows will appear only on areas covered by the :ref:`Light2D <class_Light2D>` and 
+the direction will be based on the center of the :ref:`Light <class_Light2D>`.
+
+Lights
+------
+
+:ref:`Lights <class_Light2D>` cover the entire extent of their respective Texture. What they do is use additive 
+blending to add the color of their texture to the scene. 
+
+.. image:: img/light_shadow_light.png
+
+:ref:`Lights <class_Light2D>` have four ``modes``: ``add``, ``sub``, ``mix``, and ``mask``.
+
+``Add`` adds the color of the light texture to the scene. It brightens the area under the light.
+
+``Sub`` subtracts the color of the light from the scene. It darkens the area under the light.
+
+``Mix`` mixes the color of the light with the underlying scene. The resulting brightness is
+halfway between the color of the light and the color underneath.
+
+``Mask`` is used to mask out areas that are covered by the light. Masked out areas are hidden or revealed based on 
+the color of the light.
+
+For the demo the lights have two components, the :ref:`Light <class_Light2D>` itself (which 
+is the effect of the light), and a :ref:`Sprite <class_Sprite>` blob which is just an image that shows the 
+location of the light source. A child :ref:`Sprite <class_Sprite>` is not necessary to make a 
+:ref:`Light <class_Light2D>` work but is used to show the location of the light.
+
+.. image:: img/light_shadow_light_blob.png
+
+Shadows
+-------
+
+Shadows are made by intersecting a :ref:`Light <class_Light2D>` with a :ref:`LightOccluder2D <class_LightOccluder2D>`.
+
+By default shadows are turned off. To turn them on click on the :ref:`Light <class_Light2D>` 
+and under the Shadows section check ``Enabled``.
+
+In the demo we are using a :ref:`Sprite <class_Sprite>` with a Texture on it to make the "Shadow Casters", 
+but in reality all you need is a couple :ref:`LightOccluder2Ds <class_LightOccluder2D>`. By itself 
+the :ref:`LightOccluder2D <class_LightOccluder2D>` looks like a dark spot and in this demo the :ref:`Sprite <class_Sprite>` is 
+just a black square.
+
+Step by step
+------------
+
+Now that we have covered the basics of the nodes that will be used, we will now walk step by step through 
```

Comment:
"the nodes that will be used, we will now walk step by step" -> "the nodes being used, we can walk step by step"